Adding Guid check for Reservations cmdlet arguments

corquiri opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

Adding Guid check for Reservations cmdlet arguments

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

cormacpayne commented on src/ResourceManager/Reservations/ChangeLog.md:
```diff
@@ -20,6 +20,10 @@
 -->
 ## Current Release
 
+## Version 0.1.2
```
@corquiri you can remove the version header, the text content you're adding should go under the `## Current Release` header

cormacpayne commented on src/ResourceManager/Reservations/Commands.Reservations/Common/AzureReservationsCmdletBase.cs:
```diff
@@ -21,6 +22,15 @@ public abstract class AzureReservationsCmdletBase : AzureRMCmdlet
 
         private Dictionary<string, List<string>> _defaultRequestHeaders;
 
+        protected void ValidateGuidInput(string inputValue, string inputType)
```
@corquiri is there any reason you can't just make the parameters of type `Guid` rather than of type `string`? If the server was never able to handle a non-Guid string before, then this won't be a breaking change. This will allow you to remove all of the validation you are trying to do over the strings.

corquiri replied:
Do you mean if I change parameter to something like
         [Parameter(Mandatory = false)]
         [ValidateNotNull]
         public Guid SubscriptionId { get; set; }

Then the powershell console will just throw if the input is not a GUID? Where is the code that parses the input from console and populate the parameter of cmdlet classes? If I can verify that the parameter validation is already being done, I can follow your suggestion and change the parameters to Guid type.

cormacpayne replied:
@corquiri the PowerShell engine (not controlled by us) automatically attempts to cast the provided object to whatever type the parameter is expecting, and if it's unable to do so, it will throw an error. In this case, if the user provides a string that is not in the form of a GUID, PowerShell will throw with an error saying that it is unable to convert the given string into a GUID.

maddieclayton commented:
@corquiri Ping - please address the comments above.

corquiri commented:
Reverted Guid check function and just changed the type and references accordingly.
Question : in MergeReservation.cs command, there is this parameter.
[Parameter(ParameterSetName = Constants.ParameterSetNames.CommandParameterSet, 
            Mandatory = true)]
        [ValidateNotNull]
        [ValidateCount (2, 2)]
        public Guid[] ReservationId { get; set; }

When I call merge-azurermreservation without providing this paramter, PS prompts me to enter values for this parameter (array of Guid). However, even when I specify that the exact count of this array is 2, PS keeps asking for more values after I enter 2 values. Screenshot is attached.

![capture](https://user-images.githubusercontent.com/31109331/38396600-c22f2380-38ed-11e8-8809-9bfb6d45a776.PNG)

Is there any way to avoid this situation?

cormacpayne commented:
@corquiri the `ValidateCount` attribute does validation _after_ you provide a value for the given parameter, so PowerShell will not stop prompting you for values after you have provided two. To avoid this situation, once it prompts you for the third entry, you just press enter and it will stop prompting.

cormacpayne commented:
@corquiri ping

corquiri commented:
updated changelog and re-triggering CI build because previous was not found

corquiri commented:
travis-ci build is stuck
continuous-integration/travis-ci Expected — Waiting for status to be reported
anyone familiar with this issue?

cormacpayne commented:
@corquiri I found the build in Travis and kicked it off again, as well as pushed another commit to resolve the merge conflict, and GitHub was able to find the build

cormacpayne commented:
@corquiri the build is failing because the breaking change tool is flagging the change from `string` to `Guid` as a breaking change, which it isn't since previous calls using a non-Guid string would fail. To suppress these exceptions, please follow the steps [here](https://github.com/Azure/azure-powershell/blob/preview/documentation/Debugging-StaticAnalysis-Errors.md#breaking-changes).

corquiri commented:
followed the instruction and updated breaking changes file from web editor
